core/src/main/scala/besom/util/NonEmptyString.scala

```diff
@@ -3,16 +3,17 @@ package besom.util
 import scala.quoted.*
 import scala.language.implicitConversions
 
-extension (nes: NonEmptyString)
-  inline def +++(other: String): NonEmptyString = nes + other
-  inline def asString: String                   = nes
-
 opaque type NonEmptyString <: String = String
 
 object NonEmptyString:
   def apply(s: String): Option[NonEmptyString] =
     if s.isBlank then None else Some(s)
 
+  given nonEmptyStringOps: {} with
+    extension (nes: NonEmptyString)
+      inline def +++(other: String): NonEmptyString = nes + other
```

Comment:
can we name it? 3 plus signs are hard understand IMO

Reply:
`concat`? The point is that this should be just like `++` for normal strings but because of subtyping it doesn't seem possible to reuse `++`